ui/lib/core/addon/utils/client-count-utils.ts

```diff
@@ -0,0 +1,295 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { parseAPITimestamp } from 'core/utils/date-formatters';
+import { compareAsc, getUnixTime, isWithinInterval } from 'date-fns';
+
+import type ClientsVersionHistoryModel from 'vault/vault/models/clients/version-history';
+
+/*
+The client count utils are responsible for serializing the sys/internal/counters/activity API response
+The initial API response shape and serialized types are defined below.
+
+To help visualize there are sample responses in ui/tests/helpers/clients.js
+*/
+
+// add new types here
+export const CLIENT_TYPES = [
+  'acme_clients',
+  'clients', // summation of total clients
+  'entity_clients',
+  'non_entity_clients',
+  'secret_syncs',
+] as const;
+
+type ClientTypes = (typeof CLIENT_TYPES)[number];
+
+// returns array of VersionHistoryModels for noteworthy upgrades: 1.9, 1.10
+// that occurred between timestamps (i.e. queried activity data)
+export const filterVersionHistory = (
+  versionHistory: ClientsVersionHistoryModel[],
+  start: string,
+  end: string
+) => {
+  if (versionHistory) {
+    const upgrades = versionHistory.reduce((array: ClientsVersionHistoryModel[], upgradeData) => {
+      const includesVersion = (v: string) =>
+        // only add first match, disregard subsequent patch releases of the same version
+        upgradeData.version.match(v) && !array.some((d: ClientsVersionHistoryModel) => d.version.match(v));
+
+      ['1.9', '1.10'].forEach((v) => {
+        if (includesVersion(v)) array.push(upgradeData);
+      });
+
+      return array;
+    }, []);
+
+    // if there are noteworthy upgrades, only return those during queried date range
+    if (upgrades.length) {
+      const startDate = parseAPITimestamp(start) as Date;
```

Comment:
I really wanted to convert the `parseAPITimestamp` util as well but didn't get to it 😞 , so used `as` here